### PR TITLE
feat(Fun) #495 #635

### PR DESCRIPTION
**Pull request by Aure7138**
feat(self): Fast Respawn
feat(weapon): Rapid Fire (#495)
chore(code clean): util::water (#635)
    deleted util::water and the code in self godmode and vehicle_godmode, which is repeated with proof_water, and it is better to 
    wrap it as a helper function in a class than in a util

### CODE REVIEW

File: src/backend/looped/self/fast_respawn.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+namespace big
+{
+	void looped::self_fast_respawn()
+	{
+		if (!g->self.fast_respawn) return;
+        if(PED::IS_PED_DEAD_OR_DYING(self::ped, true))
+        {
+            PED::RESURRECT_PED(self::ped);
+            ENTITY::SET_ENTITY_COORDS_NO_OFFSET(self::ped, self::pos.x, self::pos.y, self::pos.z, 0, 0, 0);
+        }
```
Comment (Yimura) [inline src/backend/looped/self/fast_respawn.cpp]: Weird indents

File: src/backend/looped/weapons/rapid_fire.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        if(PED::IS_PED_IN_ANY_VEHICLE(self::ped, 0) || HUD::IS_PAUSE_MENU_ACTIVE()) return;
+        if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_ATTACK))
+        {
+            Vector3 direction = math::rotation_to_direction(CAM::GET_GAMEPLAY_CAM_ROT(0));
+            Vector3 start = CAM::GET_GAMEPLAY_CAM_COORD() + direction;
+            Vector3 end = start + direction * 200.0;
+            Hash weapon_hash;
+		    WEAPON::GET_CURRENT_PED_WEAPON(self::ped, &weapon_hash, false);
+            MISC::SHOOT_SINGLE_BULLET_BETWEEN_COORDS(start.x, start.y, start.z, end.x, end.y, end.z, WEAPON::GET_WEAPON_DAMAGE(weapon_hash, 0), true, weapon_hash, self::ped, true, false, -1.0);
+        }
```
Comment (Yimura) [inline src/backend/looped/weapons/rapid_fire.cpp]: Weird indents

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/backend/looped/weapons/rapid_fire.cpp (hunk 2 above)
Comment (Aure7138) [reply src/backend/looped/weapons/rapid_fire.cpp]: fix

File: src/backend/looped/self/fast_respawn.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	{
+		if (g->self.fast_respawn)
+        {
+            if(PED::IS_PED_DEAD_OR_DYING(self::ped, true))
+            {
+                PED::RESURRECT_PED(self::ped);
+                ENTITY::SET_ENTITY_COORDS_NO_OFFSET(self::ped, self::pos.x, self::pos.y, self::pos.z, 0, 0, 0);
+            }
+        }
+	}
```
Comment (Yimura) [inline src/backend/looped/self/fast_respawn.cpp]: Mixed indenting, tabs and soft tabs (spaces)

File: src/backend/looped/weapons/rapid_fire.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                    Vector3 direction = math::rotation_to_direction(CAM::GET_GAMEPLAY_CAM_ROT(0));
+                    Vector3 start = CAM::GET_GAMEPLAY_CAM_COORD() + direction;
+                    Vector3 end = start + direction * 200.0;
+                    Hash weapon_hash;
+                    WEAPON::GET_CURRENT_PED_WEAPON(self::ped, &weapon_hash, false);
+                    MISC::SHOOT_SINGLE_BULLET_BETWEEN_COORDS(start.x, start.y, start.z, end.x, end.y, end.z, WEAPON::GET_WEAPON_DAMAGE(weapon_hash, 0), true, weapon_hash, self::ped, true, false, -1.0);
+                }
+            }
+        }
+	}
```
Comment (Yimura) [inline src/backend/looped/weapons/rapid_fire.cpp]: Mixed indenting, tabs and soft tabs (spaces)

**Review by Yimura — CHANGES_REQUESTED** [review]

**Comment by TheGreenBandit** [comment]
Why have it if they arent in a vehicle? Why not in a vehicle too?

**Comment by Aure7138** [comment]
> Why have it if they arent in a vehicle? Why not in a vehicle too?

Do you mean why add the rapid fire feature because you don't look like you want to have it in or out of the car
But that's your open issue lmao

> Why not in a vehicle too?

Personal preference, I added this function mainly to be able to shoot RPG quickly, and RPG cannot be used in the car, so

File: src/backend/looped/weapons/rapid_fire.cpp (hunk 4 above)
Comment (Aure7138) [reply src/backend/looped/weapons/rapid_fire.cpp]: fix

**Review by Yimura — DISMISSED** [review]

**Comment by TheGreenBandit** [comment]
> > Why have it if they arent in a vehicle? Why not in a vehicle too?
> 
> Do you mean why add the rapid fire feature because you don't look like you want to have it in or out of the car But that's your open issue lmao
> 
no i mean the same question i asked after, should've used a comma, grammatical mistake on my end

**Comment by TheGreenBandit** [comment]
but i would just have it changed so people can use it in a vehicle too because you are forgetting that there aare other weapons to be used in vehicles.

**Review by Yimura — APPROVED** [review]

**Pull request merged**